src/bigunit.ts

```diff
@@ -399,27 +410,54 @@ export class BigUnit {
     // If the length is less than the precision, pad with zeros
     if (length < this.precision) {
       const padding = "0".repeat(this.precision - length);
-      return `0.${padding}${valueString}`;
+      return isNegative 
+        ? `-0.${padding}${valueString}` 
+        : `0.${padding}${valueString}`;
     }
 
     // If the length is greater than the precision, slice the string
     const decimalIndex = length - this.precision;
     const integerPart = valueString.slice(0, decimalIndex);
     const fractionalPart = valueString.slice(decimalIndex);
-    return `${integerPart}.${fractionalPart}`;
+
+    // Construct the final string
+    const result = `${integerPart === "" ? "0" : integerPart}.${fractionalPart}`;
+    return isNegative ? `-${result}` : result;
   }
 
   public toBigInt(): bigint {
     return this.value;
   }
 
   /**
-   * @description Format the value as a string with the given precision
-   * @param precision
+   * @description Format the value as a string with the given precision (Note: this uses standard rounding and is overflow safe)
+   * @param precision - the number of digits to appear after the decimal point
    * @returns string representation of the unit value
    */
-  public format(precision: number): string {
-    return this.toNumber().toFixed(precision);
+  public format(targetPrecision: number): string {
+    BigUnit.validatePrecision(targetPrecision);
+    let scaledValue = this.value * BigInt(10n ** BigInt((Math.max(targetPrecision - this.precision, 0))));
+
+    // Apply rounding when scaling down
+    if (this.precision > targetPrecision) {
+        const divisor = BigInt(10 ** (this.precision - targetPrecision));
+        const halfDivisor = divisor / BigInt(2);
+        const remainder = scaledValue % divisor;
+        scaledValue = (scaledValue / divisor) + (remainder >= halfDivisor ? BigInt(1) : BigInt(0));
+    }
+
+    let stringValue = scaledValue.toString();
+
+    // Insert decimal point for non-zero target precision
+    if (targetPrecision > 0) {
```

Comment:
targetPrecision is already guaranteed to be > 0 because of `validatePrecision()`